src/client/dcp_client/gui/napari_window.py

```diff
@@ -22,33 +50,208 @@ def __init__(self, app: Application):
+            self.mask_choice_dropdown.addItem('Labels Mask', userData=1)
+            layout.addWidget(self.mask_choice_dropdown, 1, 1)
+
+            # when user has chosen the mask, we don't want to change it anymore to avoid errors
+            lock_button = QPushButton("Confirm Final Choice")
+            lock_button.setEnabled(False)
+            lock_button.clicked.connect(self.set_active_mask)
+
+            layout.addWidget(lock_button, 1, 2)
+            self.layer.mouse_drag_callbacks.append(self.copy_mask_callback)
+            self.layer.events.set_data.connect(lambda event: self.copy_mask_callback(self.layer, event))
 
-        buttons_layout = QHBoxLayout()
 
         add_to_inprogress_button = QPushButton('Move to \'Curatation in progress\' folder')
-        buttons_layout.addWidget(add_to_inprogress_button)
+        layout.addWidget(add_to_inprogress_button, 2, 0, 1, 2)
         add_to_inprogress_button.clicked.connect(self.on_add_to_inprogress_button_clicked)
     
         add_to_curated_button = QPushButton('Move to \'Curated dataset\' folder')
-        buttons_layout.addWidget(add_to_curated_button)
+        layout.addWidget(add_to_curated_button, 2, 2, 1, 2)
         add_to_curated_button.clicked.connect(self.on_add_to_curated_button_clicked)
 
-        layout.addLayout(buttons_layout)
-
         self.setLayout(layout)
-        self.show()
+
+    def switch_controls(self, target_widget, status: bool):
+        getattr(self.qctrl, target_widget).setEnabled(status)
+
+    def switch_to_active_mask(self):
+
+        self.switch_controls("paint_button", True)
+        self.switch_controls("erase_button", True)
+        self.switch_controls("fill_button", False)
+
+        self.active_mask = True
+    
+    def switch_to_non_active_mask(self):
+
+        self.instances = set(np.unique(self.layer.data[self.active_mask_index])[1:])
+
+        self.switch_controls("paint_button", False)
+        self.switch_controls("erase_button", False)
+        self.switch_controls("fill_button", True) 
+
+        self.active_mask = False
+
+    def set_active_mask(self):
+        self.mask_choice_dropdown.setDisabled(True)
+        self.active_mask_index = self.mask_choice_dropdown.currentData()
+        self.instances = set(np.unique(self.layer.data[self.active_mask_index])[1:])
+        self.prev_mask = self.layer.data[self.active_mask]
+        if self.active_mask_index == 1:
+            self.switch_to_non_active_mask()
+
+    def find_edges(self, idx=None):
+        '''
+        idx - indices of the specific labels from which to get contour
+        '''
+        if idx is not None and not isinstance(idx, list):
+            idx = [idx]
+
+        active_mask = self.layer.data[self.active_mask_index]
+
+        instances = np.unique(active_mask)[1:]
+        edges = np.zeros_like(active_mask).astype(int)
+
+        # to merge the discontinuous contours
+        kernel = np.ones((5, 5))
+
+        if len(instances):
```

Comment:
would it make sense instead of doing this to use ```cellpose.utils.masks_to_edges``` or ```cellpose.utils.masks_to_outlines```? Not sure which one does what we want!
https://cellpose.readthedocs.io/en/latest/api.html#cellpose.utils.masks_to_edges